packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart

```diff
@@ -0,0 +1,193 @@
+  /// user stops dragging the scroll view.
+  ///
+  /// Defaults to matching platform conventions.
+  final ScrollPhysics physics;
+
+  /// {@macro flutter.rendering.wheelList.diameterRatio}
+  ///
+  /// Defaults to an arbitrary but aesthetically reasonable number of 2.0.
+  ///
+  /// Must not be null and must be positive.
+  final double diameterRatio;
+
+  /// {@macro flutter.rendering.wheelList.perspective}
+  ///
+  /// Defaults to an arbitrary but aesthetically reasonable number of 0.003.
+  /// A larger number brings the vanishing point closer and a smaller number
+  /// pushes the vanishing point further.
+  final double perspective;
+
+  /// Size of all children in the main axis. Must not be null and must be
+  /// positive.
+  final double itemExtent;
+
+  /// Whether to clip painted children to the inside of this scroll view.
+  ///
+  /// Defaults to [true]. Must not be null.
+  ///
+  /// If this is false and [renderChildrenOutsideViewport] is false, the
+  /// first and last children may be painted partly outside of this scroll view.
+  final bool clipToSize;
+
+  /// {@macro flutter.rendering.wheelList.diameterRatio}
+  ///
+  /// Defaults to [false]. Must not be null. Cannot be true if [clipToSize]
+  /// is also true since children outside the viewport will be clipped, and
+  /// therefore cannot render children outside the viewport.
+  final bool renderChildrenOutsideViewport;
+
+  /// List of children to scroll on top of the cylinder.
+  final List<Widget> children;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scrollable(
+      controller: controller,
+      physics: physics,
+      viewportBuilder: (BuildContext context, ViewportOffset offset) {
+        return new ListWheelViewport(
+          diameterRatio: diameterRatio,
+          perspective: perspective,
+          itemExtent: itemExtent,
+          clipToSize: clipToSize,
+          renderChildrenOutsideViewport: renderChildrenOutsideViewport,
+          offset: offset,
+          children: children,
+        );
+      },
+    );
+  }
+}
+
+/// A viewport showing a subset of children on a wheel.
+///
+/// Typically used with [ListWheelScrollView], this viewport is similar to
+/// [Viewport] in that it shows a subset of children in a scrollable based
+/// on the scrolling offset and the childrens' dimensions. But uses
+/// [RenderListWheelViewport] to display the children on a wheel.
+///
+/// See also:
+///
+///  * [ListWheelScrollView], widget that combines this viewport with a scrollable.
+///  * [RenderListWheelViewport], the render object that renders the children
+///    on a wheel.
+class ListWheelViewport extends MultiChildRenderObjectWidget {
+  ListWheelViewport({
+    Key key,
+    this.diameterRatio,
+    this.perspective,
+    this.itemExtent,
+    this.clipToSize,
+    this.renderChildrenOutsideViewport,
+    this.offset,
+    List<Widget> children,
+  }) : super(key: key, children: children);
```

Comment:
why no asserts? this should surely have the same asserts as both the widget above and the render object

Reply:
Done